t/08_trento.t

```diff
@@ -41,14 +48,115 @@ subtest '[k8s_logs] Get logs from running pods as it is also required' => sub {
+            # {"assets": {"sources": [{"format": "tar.gz", "url": "https://gitlab.suse.de/qa-css/trento/-/archive/v0.2.0/trento-v0.2.0.tar.gz"}]}}
+            if ($_[0] =~ /curl.*api\/v4\/projects\/$gitlab_prj_id\/releases\/v$gitlab_ver.*jq.*assets/) { return $gitlag_tar; }
+            if ($_[0] =~ /curl.*api\/v4\/projects\/$gitlab_prj_id\/releases\/v$gitlab_ver.*jq.*commit/) { return $gitlab_sha; }
+            return '';
+    });
+    set_var('TRENTO_DEPLOY_VER', $gitlab_ver);
+    set_var('_SECRET_TRENTO_GITLAB_TOKEN', $password);
+    get_trento_deployment('self', '/tmp');
+    note(join("\n  1C-->  ", @calls));
+    like $recorded_passwords[0], qr/$password/;
+    ok any { /curl.*api\/v4\/projects\/$gitlab_prj_id\/repository\/.*sha=$gitlab_sha.*--output $gitlag_tar/ } @calls;
+    ok any { /tar.*$gitlag_tar/ } @calls;
+};
+
+subtest '[get_trento_ip] check the az command' => sub {
+    @calls = ();
+    $trento->redefine(get_current_job_id => sub { return 'PASTICCINO'; });
+    $trento->redefine(script_output => sub { push @calls, $_[0]; return 'GINGERINO'; });
+
+    my $out = get_trento_ip();
+    note(join("\n  1C-->  ", @calls));
+    ok $out eq 'GINGERINO';
+    like $calls[0], qr/az vm show -d -g .*PASTICCINO -n .*PASTICCINO --query "publicIps" -o tsv/;
 };
 
+subtest '[cypress_configs]' => sub {
+    @calls = ();
+    $trento->redefine(assert_script_run => sub { push @calls, $_[0]; return 'PATATINE'; });
+    $trento->redefine(get_trento_ip => sub { return '43.43.43.43' });
+    $trento->redefine(get_trento_password => sub { return 'SPUMA_DI_TONNO'; });
+    my $nodes = 42;
+    $trento->redefine(get_agents_number => sub { return $nodes; });
+
+    my $ver = '9.8.7';
+    set_var('TRENTO_VERSION' => $ver);
+    cypress_configs('/FESTA/BANCONE/SPREMUTA');
+    note(join("\n -->  ", @calls));
+    ok any { /cypress\.env\.py -u .*43\.43\.43\.43 -p SPUMA_DI_TONNO -f Premium -n $nodes --trento-version $ver/ } @calls;
+};
+
+subtest '[get_agents_number]' => sub {
+    my $cp = 'POLPETTE';
+    set_var('PUBLIC_CLOUD_PROVIDER' => $cp);
+    set_var('QESAP_CONFIG_FILE' => 'MELANZANE_FRITTE');
+
+
+    my $tmp_folder = tempdir(CLEANUP => 1);
+    note("-->tmp_folder=$tmp_folder");
+    set_var('QESAP_DEPLOYMENT_DIR' => $tmp_folder);
+
+
+    my $inv_path = "$tmp_folder/terraform/" . lc $cp;
+    make_path($inv_path, {verbose => 1});
+
+    my $str = <<END;
+all:
+  children:
+    hana:
+      hosts:
+        vmhana01:
+          ansible_host: 1.2.3.4
+          ansible_python_interpreter: /usr/bin/python3
+        vmhana02:
+          ansible_host: 1.2.3.5
+          ansible_python_interpreter: /usr/bin/python3
+
+    iscsi:
+      hosts:
+        vmiscsi01:
+          ansible_host: 1.2.3.6
+          ansible_python_interpreter: /usr/bin/python3
+
+  hosts: null
+END
+    open(FH, '>', $inv_path . '/inventory.yaml') or die $!;
+    print FH $str;
+    close(FH);
```

Comment:
Using a so-called bareword filehandle is very old perl. the filehandles are global, and in the worst case different code using the same filehandle can conflict.
Since perl 5.6 one can use lexical filehandles like normal variables:
```suggestion
    open my $fh, '>', $inv_path . '/inventory.yaml' or die $!;
    print $fh $str;
    close $fh;
```